Makefile

```diff
@@ -93,6 +93,14 @@ generate:
 generate-guardrails:
 	cd pkg/operator/controllers/guardrails/policies && ./scripts/generate.sh > /dev/null
 
+generate-kiota:
```

Comment:
idea: can we add this as a dependent target of `generate` it feels like a good fit there, and would mean it gets tied into all our CI/etc.

Reply:
I don't think this is a good idea, as the Kiota generator is something that's probably a bit annoying to install and will need to actually be run every once in a blue moon, so I just added some docs for when it's needed instead.